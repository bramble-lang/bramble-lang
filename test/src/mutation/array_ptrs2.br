fn my_main() -> i64 {
    let mut x: i64 := 1;
    let mut y: i64 := 2;
    let mut arr: [*mut i64; 2] := [@mut x, @mut y];
    project::std::io::writei64ln(^arr[0]); // 1
    project::std::io::writei64ln(^arr[1]); // 2

    mut ^arr[0] := 13;
    project::std::io::writei64ln(^arr[0]); // 13
    project::std::io::writei64ln(^arr[1]); // 2

    mut y := 23;
    project::std::io::writei64ln(^arr[0]); // 13
    project::std::io::writei64ln(^arr[1]); // 23

    let mut z: i64 := 3;
    mut arr[0] := @mut z;
    project::std::io::writei64ln(x); // 13
    project::std::io::writei64ln(^arr[0]); // 3
    project::std::io::writei64ln(^arr[1]); // 23

    mut ^arr[0] := 37;
    project::std::io::writei64ln(x); // 13
    project::std::io::writei64ln(^arr[0]); // 37
    project::std::io::writei64ln(^arr[1]); // 23

    return 0;
}