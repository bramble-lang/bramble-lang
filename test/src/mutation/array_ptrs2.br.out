1
2
13
2
13
23
13
3
23
13
37
23
